### PR TITLE
Simplify, cleanup, update code

### PR DESCRIPTION
**Pull request by okurz**
* Simplify loader/qem
* Simplify smeltsync
* Extract 'handle_arch' method in types/aggregates
* Simplify counter assignment in aggregates
* tests: Simplify test_repohash
* tests: Simplify remaining approver generators
* tests: Extract 'approver' helper in test_approve
* tests: Extract "two_passed_responses" helper
* Simplify tests
* tests: Skip 'responses' test if not available on older python
* Use updated codecov uploader v3
* approver: Extract _handle_http_error function
* Squeeze some whitespace in code with shorter 'log'
* approver: Extract approvable method
* Add and change log messages in case of approval-blocking tests

### CODE REVIEW

**Review by Martchus — APPROVED** [review]
Reviewed it commit-by-commit. I trust that the "Extract …" commits are really just moving the code around.

**Review by baierjan — APPROVED** [review]

**Comment by okurz** [comment]
rebased after conflict resolution

File: openqabot/approver.py [hunk 1]
```diff
@@ -112,12 +135,16 @@ def get_jobs(self, job: JobAggr, api: str, inc: int) -> bool:
...
             if self.is_job_marked_acceptable_for_incident(job, inc):
-                logger.info(
+                log.info(
                     "Ignoring failed job %s for incident %s due to openQA comment"
                     % (job.job_id, inc)
                 )
                 res["status"] = "passed"
             else:
+                log.info(
+                    "Found failed, not-ignored job %s for incident %s"
```
Comment (perlpunk) [inline openqabot/approver.py:145]: This only logs the first failed job found.

File: openqabot/approver.py [hunk 2]
```diff
@@ -37,58 +58,60 @@ def __init__(self, args: Namespace) -> None:
...
+
+            if any(i.withAggregate for i in i_jobs):
+                log.info("Aggregate missing for %s" % str(inc.inc))
+                return False
+
+            u_jobs = []
+
+        if not self.get_incident_result(i_jobs, "api/jobs/incident/", inc.inc):
+            log.info(
+                "Inc %s has at least one failed job in incident tests" % str(inc.inc)
```
Comment (perlpunk) [inline openqabot/approver.py:103]: Why the "at least" here and not in line 109?

File: openqabot/approver.py (hunk 1 above)
Comment (okurz) [reply openqabot/approver.py:145]: correct, only the first

File: openqabot/approver.py (hunk 2 above)
Comment (okurz) [reply openqabot/approver.py:103]: I think I just missed that. Thx.

Created https://github.com/openSUSE/qem-bot/pull/87

**Pull request merged**